### ISSUE TITLE
[BUG] openapi-generator mvn plugin - typeMappings does not appear to take hold in individual executions

### ISSUE THREAD

**Issue by randeepbydesign** [OpenAPITools/openapi-generator#11026]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [ ] Have you searched for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

We use the maven plugin to generate our artefacts.
I was setting up type mappings in my executions so that int32 types were mapped to long instead of Integer:

```
    <execution>
      <id>myserver-me-api</id>
      <goals>
        <goal>generate</goal>
      </goals>
      <configuration>
        <inputSpec>${project.basedir}/src/main/resources/myserver-me-api.json</inputSpec>
        <generateApiTests>false</generateApiTests>
        <generateModelTests>false</generateModelTests>
        <generatorName>java</generatorName>
        <configOptions>
          <sourceFolder>com/myserver/api</sourceFolder>
          <apiPackage>client.myserver</apiPackage>
          <modelPackage>client.myserver.model</modelPackage>
          <invokerPackage>client.myserver.handler</invokerPackage>
          <library>resttemplate</library>
          <java8>true</java8>
          <importMappings>integer=Long</importMappings>
          <typeMappings>integer=Long,int=Long</typeMappings>
          <dateLibrary>java8-localdatetime</dateLibrary>
          <serializableModel>true</serializableModel>
        </configOptions>
      </configuration>
    </execution>
```

However it had no effect and my generated POJO still used Integer here. However, defining typeMappings in the `plugin` section did work:

```
  <plugin>
    <groupId>org.openapitools</groupId>
    <artifactId>openapi-generator-maven-plugin</artifactId>
    <version>5.1.0</version>
    <configuration>
      <typeMappings>integer=Long,int=Long</typeMappings>
    </configuration>
    <executions>
        ...
```

Is this a feature-not-a-bug kind of thing or are we supposed to be able to configure the individual executions this way? The `openapi-generator-maven-plugin` markdown document is not very clear on the distinction so I am unsure.

##### openapi-generator version
Using 5.1.0 of openapi-generator-maven-plugin

##### Steps to reproduce

<build>
<plugins>
<plugin>
<groupId>org.openapitools</groupId>
<artifactId>openapi-generator-maven-plugin</artifactId>
<version>5.1.0</version>
<configuration>
	<typeMappings>integer=Long,int=Long</typeMappings> 
</configuration>
<executions>
	<execution>
	<id>myapp-me-api</id>
	<goals>
		<goal>generate</goal>
	</goals>
	<configuration>
		<inputSpec>${project.basedir}/src/main/resources/myapp-me-api.json</inputSpec>
		<generateApiTests>false</generateApiTests>
		<generateModelTests>false</generateModelTests>
		<generatorName>java</generatorName>
		<configOptions>
		<sourceFolder>com/myapp/api</sourceFolder>
		<apiPackage>client.myapp</apiPackage>
		<modelPackage>client.myapp.model</modelPackage>
		<invokerPackage>client.myapp.handler</invokerPackage>
		<library>resttemplate</library>
		<java8>true</java8>
		<typeMappings>integer=Long,int=Long</typeMappings> 
		<dateLibrary>java8-localdatetime</dateLibrary>
		<serializableModel>true</serializableModel>
		</configOptions>
	</configuration>
	</execution>
</executions>
</plugin>
</plugins>
</build>

**Comment by borsch** [comment]
@randeepbydesign try to use
```
<importMappings>
  <importMapping>integer=Long</importMapping>
</importMappings>
```
This works for me on plugin level. Haven't checked on execution level, but who knows :)

**Comment by randeepbydesign** [comment]
Yes- if you look at my description I confirm that it works at plugin level but _not_ at execution level.

**Comment by randeepbydesign** [comment]
Ah I see what you were doing. Yes, a specific `importMapping` does not work at the execution level either

**Comment by zeekoe** [comment]
Wow, thanks for posting this. I also expected this in the configOptions, it took me nearly an hour to find this. Perhaps the documentation [here](https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator-maven-plugin) can be extended with an example where to put the typeMappings (and the fact that importMappings does not work anymore)?

**Comment by mfruBRZ** [comment]
In case it helps someone else:

I started out with 

```xml 
<typeMappings><typeMapping>OffsetDateTime=Instant</typeMapping></typeMappings>
<importMappings><importMapping>java.time.OffsetDateTime=java.time.Instant</importMapping></importMappings>
```

in the execution configuration which was not picked up.

Took a look at @zeekoe link to the docs and noticed that the format is type=path, so I changed it to

```xml
<typeMappings><typeMapping>OffsetDateTime=java.time.Instant</typeMapping></typeMappings>
<importMappings><importMapping>OffsetDateTime=java.time.Instant</importMapping></importMappings>
```

which eventually worked for me.
